### ISSUE TITLE
Implementation for Squashed Gaussian

### ISSUE THREAD

**Issue by jy-cds** [deepmind/rlax#11]
Hi! I have questions regarding the implementation for squashed gaussian distribution in distribution.py
1) In https://github.com/deepmind/rlax/blob/b7d1a012f888d1744245732a2bcf15f38bb7511e/rlax/_src/distributions.py#L319, should it be "-=" instead of the current "+="? because we have to take log of the inverse of determinant?

2) I am having trouble understanding a particular function in the implementation of the squashed gaussian in:
https://github.com/deepmind/rlax/blob/b7d1a012f888d1744245732a2bcf15f38bb7511e/rlax/_src/distributions.py#L295

In particular, I dont understand why the  the mean (_mu_) and standard deviation (_sigma_) of the
original Gaussian random variable (pre-squash) have to be "activated". The mean (_mu_) seems to go through a "tanh" transformation and variable _sigma_ goes through a really odd-looking transformation on line 295 and 299. Why do we have to need _sigma_min_(default to -4) and _sigma_max_(default to 0)? What is the meaning of these transformations on _mu_ and _sigma_? 

3) My understanding of the implementation is that instead of "squashing" a gaussian random variable (call it X) into (-1,1), we are squashing X into specific action_specs from the environment. Therefore, the transformation/squashing function is no long just Y = tanh(X) as in the original SAC paper but rather Y = g(X) = scale * (tanh(X) + 1) +min_vals and this is confirmed by the function "transform": https://github.com/deepmind/rlax/blob/b7d1a012f888d1744245732a2bcf15f38bb7511e/rlax/_src/distributions.py#L302. I went through the math to derive the pdf and logpdf of this new transformed/squashed random variable Y, and my result is nothing like the code. Did i miss something in the code? I suspect that this is related to question 2. Any comment and reference pointers would be much appreciated!

**Comment by katebaumli** [comment]
Hello, thanks for your question!

1. Notice that we subtract the value that is returned from `log_det_jacobian` in line 355, so the overall sum of this expression does indeed correspond to the log of the inverse of determinant. The concrete way it is implemented here is for numerical stability.

2. The activation on the standard deviation is in order to keep it bounded between these values. It is not necessary to have this transformation nor to set it to these default values, but it is the way we modeled it in this function. The extra `tanh` activation on the mean was not used in the original sac paper but we found it to be useful in some agents. Feel free to remove it.

**Comment by fuyw** [comment]
Hello, I have two questions about the `logprob_fn`:

1. Should we have a 0.5 in `half_logdet = jnp.sum(jnp.log(sigma), axis=-1)`?

2. Should we use `inv_transform(sample, action_spec)` instead of `sample` in the `log_det_jacobian(sample, action_spec)`?
